### PR TITLE
Typechecker Cleanup

### PR DESCRIPTION
**Pull request by eliaslfox**

### CODE REVIEW

**Review by stylewarning — APPROVED** [review]
I had one question.

My only other comment is I'm not a fan of the `-g` suffix. I don't know what it means.

File: src/typechecker/kinds.lisp [hunk 1]
```diff
@@ -44,48 +46,46 @@
...
 
-#+(and sbcl coalton-release)
-(declaim (sb-ext:freeze-type kind))
 
-(defvar *next-kvar-id* 0)
+;;;
+;;; Kind Variables
+;;;
+
+(defparameter *next-kvar-id* 0)
```
[the remark this answers is not in the document]
Comment (eliaslfox) [reply src/typechecker/kinds.lisp:88]: Yes. I got those backwards.

**Comment by eliaslfox** [comment]
`-g` means generic. Some of the functions use clos to traverse structures, and then have a single remove-duplicates call in the top function. The general pattern is:

```
(defun function (x)
  (remove-duplicates (function-g x) :test #'eq))

(defgeneric function-g (x))
```

Callers aren't intended to call the `-g` version directly. Is there a way I could make this more clear?

**Comment by gefjon** [comment]
> Callers aren't intended to call the -g version directly. Is there a way I could make this more clear?

You could extend `-g` to `-generic`, and prepend a `%` to denote that it's not a public interface.

**Pull request merged**